extensions/replication/queueProcessor/MultipleBackendTask.js

```diff
@@ -183,24 +184,17 @@ class MultipleBackendTask extends QueueProcessorTask {
         }, cb);
     }
 
-    _getAndPutMultipartUploadOnce(sourceEntry, destEntry, log, cb) {
-        const doneOnce = jsutil.once(cb);
-        log.debug('replicating MPU data', { entry: sourceEntry.getLogInfo() });
-        if (sourceEntry.getLocation().some(part => {
-            const partObj = new ObjectMDLocation(part);
-            return partObj.getDataStoreETag() === undefined;
-        })) {
-            log.error('cannot replicate object without dataStoreETag property',
-                {
-                    method: 'MultipleBackendTask._getAndPutMultipartUploadOnce',
-                    entry: sourceEntry.getLogInfo(),
-                });
-            return cb(errors.InvalidObjectState);
+    _initiateMPU(sourceEntry, destEntry, log, cb) {
+        const storageType = destEntry.getReplicationStorageType();
+        // If using Azure backend, create a unique ID to use as the block ID.
+        if (storageType === 'azure') {
+            const uploadId = UUID.v4().replace(/-/g, '');
+            return cb(null, uploadId);
```

Comment:
use `setImmediate()` to ensure an asynchronous context in all cases and avoid potential annoying and hard to debug errors.

Reply:
Thanks for this. Updated.